src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -506,3 +513,40 @@ mongoc_client_pool_enable_auto_encryption (mongoc_client_pool_t *pool,
    return _mongoc_cse_client_pool_enable_auto_encryption (
       pool->topology, opts, error);
 }
+
+bool
+mongoc_client_pool_set_timeout_ms (mongoc_client_pool_t *pool,
+				   int64_t timeout,
+				   bson_error_t *error) {
+   if (!pool) {
```

Comment:
Checking that the class parameter in class methods is not NULL isn't generally something we do at all in the driver right now.  So if a user uses most of those methods with a NULL client/pool/db/coll, the program could seg fault or return garbage of some kind (see `mongoc_client_pool_enable_auto_encryption` just above this method).  

This seemed like a much more graceful way to handle that situation.  If you like it, then perhaps it could become our convention for methods like this.

Reply:
Isn't `BSON_ASSERT` commonly used in other class methods? Despite the name, it doesn't use the `assert()` (which could be compiled out). On failure, it will `fprintf` to `stderr` and `abort()`.

I see some other pool methods use `BSON_ASSERT`, so maybe this was just missed in `mongoc_client_pool_enable_auto_encryption`.

Reply:
Found `BSON_ASSERT_PARAM`, which seems to be made for this purpose!